### PR TITLE
revamp game-catalog-starter

### PR DESCRIPTION
**Pull request by GrosboillotFlo**
## Description

Related to this [revamp](https://github.com/fewlinesco/sparta_courses/pull/908) :

  - `README.md` : add yarn commands for mongoDB `populate-db` and `destroy-db` scripts
  - `.env.sample` : add sample for required environemental variables
  - `package.json` : bumps, except for `node-fetch` because ES6 🤷 
  - `tool-versions` : not sure about this one, maybe because of my local setup with yarn and node ?
  - `tests/e2e/loading.test.ts` : mainly adding a `setTimeout()`, it seems that the connection to a mongo database take enough time to be reject by default timeout of jest.

## Related Issue

[CU-1mmprzg](https://app.clickup.com/t/1mmprzg)

## Motivation and Context

As we went from Connect to Auth0, we had to update this starter project.

## How Has This Been Tested?

Tested with `yarn test`

## Types of changes

- Chore (non-breaking change which refactors / improves the existing code base)
- ~Bug fix~ (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to 
  change)

## Checklist:

- :white_check_mark: My code follows the code style of this project.
- :white_check_mark: My change requires a change to the documentation.
- :white_check_mark: I have updated the documentation accordingly.
- :white_check_mark: I have read the [**CONTRIBUTING**][CONTRIBUTING_FILE] document.
- :red_circle: I have added tests to cover my changes.
- :white_check_mark: All new and existing tests passed.

[CONTRIBUTING_FILE]: https://github.com/fewlinesco/connect/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by GrosboillotFlo** [comment]
@Fenntasy @TheRealAstoo : as mentioned in the description, the connection to mongoDB seems really slow, enough to have the tests fail. Am I missing something on it ? Does it mean that we will have to check on other exercises/tests ? 🤔

File: .env.sample [hunk 1]
```diff
@@ -1,2 +1,9 @@
...
-MONGO_URL=
+PORT= 3000
+MONGO_URL=<YOUR_MONGO_DB_URL>
+HOST=<YOUR_LOCAL_HOST_URL>
+DOMAIN=https://<YOUR_AUTH0_DOMAIN>
+CLIENTID=<YOUR_AUTH0_CLIENT_ID>
+CLIENTSECRET=<YOUR_AUTH0_CLIENT_SECRET>
+REDIRECTURI=<YOUR_CALLBACK_URL>
+AUDIENCE=https://<YOUR_AUTH0_DOMAIN>/api/v2/
+SCOPES= openid profile email offline_access
```
Comment (Fenntasy) [inline .env.sample]:
```suggestion
SCOPES="openid profile email offline_access"
```

File: .env.sample [hunk 2]
```diff
@@ -1,2 +1,9 @@
-PORT=3000
-MONGO_URL=
+PORT= 3000
+MONGO_URL=<YOUR_MONGO_DB_URL>
+HOST=<YOUR_LOCAL_HOST_URL>
+DOMAIN=https://<YOUR_AUTH0_DOMAIN>
+CLIENTID=<YOUR_AUTH0_CLIENT_ID>
+CLIENTSECRET=<YOUR_AUTH0_CLIENT_SECRET>
+REDIRECTURI=<YOUR_CALLBACK_URL>
```
Comment (Fenntasy) [inline .env.sample]:
```suggestion
CLIENT_ID=<YOUR_AUTH0_CLIENT_ID>
CLIENT_SECRET=<YOUR_AUTH0_CLIENT_SECRET>
REDIRECT_URI=<YOUR_CALLBACK_URL>
```

however, those are not the same name as in your course, is that wanted?

Comment (GrosboillotFlo) [reply .env.sample]: No, it's not: I will wait for [this related PR](https://github.com/fewlinesco/sparta_courses/pull/908) to be merged for changes here.

**Comment by GrosboillotFlo** [comment]
@Fenntasy It seems we need to update the image (`node:14.8-buster`) on [`unit-tests.yaml`](https://github.com/fewlines-education/game-catalog-starter/blob/main/.github/workflows/e2e-tests.yaml): updating packages on this directory makes it outdated. Any thoughts on this? 🤔 

The error during CI unit tests : 
```
Run yarn install
yarn install v1.22.4
[1/4] Resolving packages...
[2/4] Fetching packages...
error jest-diff@27.4.2: The engine "node" is incompatible with this module. Expected version "^10.13.0 || ^12.13.0 || ^14.15.0 || >=15.0.0". Got "14.8.0"
error Found incompatible module.
```

**Comment by Fenntasy** [comment]
> @Fenntasy It seems we need to update the image (`node:14.8-buster`) on [`unit-tests.yaml`](https://github.com/fewlines-education/game-catalog-starter/blob/main/.github/workflows/e2e-tests.yaml): updating packages on this directory makes it outdated. Any thoughts on this? 🤔

Well... what about `node:16.13` since it's the latest stable? Does it work with it?

File: .tool-versions [hunk 3]
```diff
@@ -1,2 +1,2 @@
-nodejs 14.13.0
-yarn 1.22.10
+nodejs 16.2.0
```
Comment (Fenntasy) [inline .tool-versions]: Why not `16.13.1`?

We simplify for the container in `unit-tests.yaml` by using `16.3` but it's still a good thing to have the same version in both environment

File: .tool-versions [hunk 4]
```diff
@@ -1,2 +1,2 @@
...
+yarn 1.22.4
```
Comment (Fenntasy) [inline .tool-versions]: 🤔 4 is < 10, why do we go back?

Comment (GrosboillotFlo) [reply .tool-versions]: Hasty copy/paste 🤦

**Review by Fenntasy — APPROVED** [review]

**Pull request merged**